starter-code/routes/index.js

```diff
@@ -1,9 +1,58 @@
 var express = require('express');
 var router = express.Router();
+var Beach = require ('../models/beach');
 
 /* GET home page. */
 router.get('/', function(req, res, next) {
-  res.render('index', { title: 'Express' });
+  res.render('index', { title: 'My Beaches' });
+});
+
+router.post('/beach', function(req, res, next) {
+  var beachName= req.body.beachName;
+  var color= req.body.flag;
+  console.log(req.body);
+
+  var beach = new Beach({
+    beachName: beachName,
+    flag: color
+  });
+
+  Beach.findOne({beachName: beachName}, function(err, doc){
+    if (err) {
+      return next(error);
+    } else {
+      if (doc) {
+        Beach.findOneAnddUpdate({beachName: beachName}, {flag: flagColor}, function(err, doc){
+          if(err){
+            return next(err);
+          } else {
+            beach.save((err), function(){
```

Comment:
Let's review the code flow. First, you search if the beach exists in the database (line 20). If there is no error and the beach exists (line 24), you update it. Then, instead of saving if the beach doesn't exist (and else to the if in line 24), you try to save it in the else of the error for the update (line 28). In conclusion, the beach would be only saved IF THERE WERE AN ERROR TRYING TO UPDATE IT.